### PR TITLE
Add ZerionBalancesSchema tests

### PR DESCRIPTION
**Pull request by hectorgomezv**
## Changes
- Adds `ZerionBalances` entity schema unit tests.
- Changes `nullable()` by `nullish().default(null)` for several fields in the `ZerionBalance` schema, following the [Zerion API documentation](https://developers.zerion.io/reference/listwalletpositions).

### CODE REVIEW

File: src/datasources/balances-api/entities/zerion-balance.entity.ts [hunk 1]
```diff
@@ -19,41 +19,41 @@ export type ZerionBalance = z.infer<typeof ZerionBalanceSchema>;
 
 export type ZerionBalances = z.infer<typeof ZerionBalancesSchema>;
 
-const ZerionImplementationSchema = z.object({
+export const ZerionImplementationSchema = z.object({
```
Comment (hectorgomezv) [inline src/datasources/balances-api/entities/zerion-balance.entity.ts:22]: The schemas in this file are exported to test them on the ad hoc tests.

File: src/datasources/balances-api/zerion-balances-api.service.ts [hunk 2]
```diff
@@ -222,9 +222,9 @@ export class ZerionBalancesApi implements IBalancesApi {
           throw Error(
             `Zerion error: ${chainName} implementation not found for balance ${zb.id}`,
           );
-        const { value } = zb.attributes;
+        const { value, price } = zb.attributes;
         const fiatBalance = value ? getNumberString(value) : null;
-        const fiatConversion = getNumberString(zb.attributes.price);
+        const fiatConversion = price ? getNumberString(price) : null;
```
Comment (hectorgomezv) [inline src/datasources/balances-api/zerion-balances-api.service.ts:227]: `attributes.price` is not marked as required in the [Zerion API docs](https://developers.zerion.io/reference/listwalletpositions), so I've changed this to reflect the possibility of the field being `undefined`.

File: src/datasources/balances-api/entities/zerion-balance.entity.spec.ts [hunk 3]
```diff
@@ -0,0 +1,557 @@
...
+
+      const result = ZerionBalanceSchema.safeParse(zerionBalance);
+
+      expect(result.success).toBe(true);
+    });
+
+    it('should not allow an invalid type value', () => {
+      const zerionBalance = zerionBalanceBuilder().build();
+      // @ts-expect-error - type is expected to be a 'positions' literal
+      zerionBalance.type = 'invalid';
```
Comment (iamacook) [inline src/datasources/balances-api/entities/zerion-balance.entity.spec.ts:64]: What do you think about adding an `unknown` literal here if it throws?

File: src/datasources/balances-api/entities/zerion-balance.entity.ts [hunk 4]
```diff
@@ -19,41 +19,41 @@ export type ZerionBalance = z.infer<typeof ZerionBalanceSchema>;
...
   chain_id: z.string(),
-  address: z.string().nullable(),
+  address: z.string().nullish().default(null),
```
Comment (iamacook) [inline src/datasources/balances-api/entities/zerion-balance.entity.ts:24]: We could use `AddressSchema` here (and add a test for checksumming).

File: src/datasources/balances-api/entities/zerion-balance.entity.spec.ts [hunk 5]
```diff
@@ -0,0 +1,557 @@
...
+            message: 'Invalid literal value, expected "positions"',
+          },
+        ]),
+      );
+    });
+
+    it('should not allow an invalid id value', () => {
+      const zerionBalance = zerionBalanceBuilder().build();
+      // @ts-expect-error - id is expected to be a string
+      zerionBalance.id = faker.number.int();
```
Comment (iamacook) [inline src/datasources/balances-api/entities/zerion-balance.entity.spec.ts:84]: I was initially going to suggest we move this into the builder but this is a much more future-proof method of overwriting the values that we should consider doing elsewhere in schema tests - if the type changes to match that expected, TypeScript will complain. (If we casted it inside the `with` of a builder, it wouldn't.)

Comment (hectorgomezv) [reply src/datasources/balances-api/entities/zerion-balance.entity.spec.ts:84]: It makes perfect sense to me! 👍🏻

Comment (hectorgomezv) [reply src/datasources/balances-api/entities/zerion-balance.entity.spec.ts:84]: It makes perfect sense to me! 👍🏻

File: src/datasources/balances-api/entities/zerion-balance.entity.spec.ts (hunk 3 above)
Comment (hectorgomezv) [reply src/datasources/balances-api/entities/zerion-balance.entity.spec.ts:65]: Added in [efadc24](https://github.com/safe-global/safe-client-gateway/pull/1637/commits/efadc242b984be3bbd2955dff58249aaf050779c)

**Review by iamacook — APPROVED** [review]

**Pull request merged**